src/test/java/edu/ucla/library/iiif/fester/utils/TestUtils.java

```diff
@@ -44,79 +56,32 @@ public static String getTestData(final Vertx aVertx, final String aPath) {
-
-            final Matcher idMatcher = idPattern.matcher(manifest.toString());
-            final StringBuilder sb = new StringBuilder();
-
-            final Map<String, String> annoPageIdReplacementMap = new HashMap<>();
-            final Map<String, String> annoIdReplacementMap = new HashMap<>();
-            final Map<String, String> canvasIdReplacementMap = new HashMap<>();
-
-            int annoPageIdMatchCount = 0;
-            int annoIdMatchCount = 0;
-            int canvasIdMatchCount = 0;
-
-            // De-randomize all the IIIF resource IDs so the manifests can be compared
-            while (idMatcher.find()) {
-                final String annoPageIdMatch = idMatcher.group("annopage");
-                final String annoIdMatch = idMatcher.group("anno");
-                final String canvasIdMatch = idMatcher.group("canvas");
-
-                final String replacementId;
-
-                if (annoPageIdMatch != null) {
-                    if (annoPageIdReplacementMap.containsKey(annoPageIdMatch)) {
-                        // Use the same de-randomized ID for instances of the same ID
-                        replacementId = annoPageIdReplacementMap.get(annoPageIdMatch);
-                    } else {
-                        // Create a new de-randomized ID
-                        replacementId = StringUtils.format("anno-page-{}", ++annoPageIdMatchCount);
-                        annoPageIdReplacementMap.put(annoPageIdMatch, replacementId);
-                    }
-                } else if (annoIdMatch != null) {
-                    if (annoIdReplacementMap.containsKey(annoIdMatch)) {
-                        replacementId = annoIdReplacementMap.get(annoIdMatch);
-                    } else {
-                        replacementId = StringUtils.format("anno-{}", ++annoIdMatchCount);
-                        annoIdReplacementMap.put(annoIdMatch, replacementId);
-                    }
-                } else { // canvas
-                    if (canvasIdReplacementMap.containsKey(canvasIdMatch)) {
-                        replacementId = canvasIdReplacementMap.get(canvasIdMatch);
-                    } else {
-                        replacementId = StringUtils.format("canvas-{}", ++canvasIdMatchCount);
-                        canvasIdReplacementMap.put(canvasIdMatch, replacementId);
-                    }
-                }
-                idMatcher.appendReplacement(sb, replacementId);
-            }
-            idMatcher.appendTail(sb);
-
-            return new JsonObject(sb.toString());
-
-        }).distinct().count();
-
-        return distinctManifestCount == 1;
+    private static JsonObject sanitize(final JsonObject aManifest, final String[] aKeyArray) {
+        String json = aManifest.encode();
+
+        for (final String name : aKeyArray) {
+            json = json.replaceAll(StringUtils.format(ID_PATTERN, name), StringUtils.format(EMPTY_ID, name));
+        }
+
+        return new JsonObject(json);
```

Comment:
I appreciate that this is a much simpler implementation, but I'm not sure that we don't need to check _any_ ids. Should we test that ids of A/V resources end with `/manifest.mpd` or `/playlist.m3u8`, for example?

Reply:
The tests where we're using this don't seem to be relevant to this. I could see an argument for testing those, but those should be more specific tests in my opinion rather than structural tests that also look at ID values.

Reply:
Not relevant even though 3 of the 4 tests we're using this on involve A/V material?

Reply:
Yes, if we want to validate A/V materials' IDs, we should have specific tests for that. Keeping the tests focused on one thing makes them more useful and easier to maintain.

Reply:
Resolution: we chatted more in Slack. I'm going to implement a check that includes the non-random parts of the IDs too here.